lib/shopify_api/session.rb

```diff
@@ -51,7 +51,7 @@ def prepare_url(url)
       end
 
       def validate_signature(params)
-        params = params.with_indifferent_access
+        params = (params.respond_to?(:to_unsafe_hash) ? params.to_unsafe_hash : params).with_indifferent_access
```

Comment:
The correct way to fix this deprecation is calling `to_h`, but that is going to show another deprecation message unless the application already fixed it or making sure that the argument of this method is a hash with indifferent access (since `Parameters` is already one and you don't need to make another one again) or a hash with symbols (since it is all we want). That being said have we tried to remove the `with_indifferent_access` call?

Reply:
Has this been abandoned? I'm running into the issue right now 😄 
In 5.2.1 it's no longer a deprecation warning, they simply removed the method 🎉 

Would it be crazy to just use `.permit!` instead?

@rafaelfranca @bdewater @andrewle

Reply:
That has the same as `to_unsafe_hash`, which BTW is the same as `permit!.to_h`. I think the proper fix would remove the `with_indifferent_access` call from here.

Reply:
I don't even think we need `.to_h`. Why would we need it?

Reply:
because `with_indifferent_access` is not defined in `Parameters` instance.

Reply:
oh, yeah. We can, but I really prefer if we don't call neither of those methods. `permit!` is not required since we access the hash using `[]` and we don't need `with_indifferent_access`.

Reply:
@rafaelfranca 
Created a new PR as I don't have access to push to this one -- I guess this one can then be closed. I can make updates to my branch or give access to people if needed.

https://github.com/Shopify/shopify_api/pull/468